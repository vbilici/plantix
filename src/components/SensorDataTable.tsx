import { FC } from "react";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeaderCell,
  TableRow,
} from "@tremor/react";

interface SensorData {
  sensorId: string;
  type: string;
  value: number;
  timestamp: string;
}

interface SensorDataTableProps {
  sensorData: SensorData[];
}

const SensorDataTable: FC<SensorDataTableProps> = ({ sensorData }) => {
  if (!Array.isArray(sensorData)) {
    return null;
  }
  return <div className="mx-auto">
    <Table >
      <TableHead>
        <TableRow>
          <TableHeaderCell>Sensor Id</TableHeaderCell>
          <TableHeaderCell>Type</TableHeaderCell>
          <TableHeaderCell className="text-right">Value</TableHeaderCell>
          <TableHeaderCell>Timestamp</TableHeaderCell>
        </TableRow>
      </TableHead>

      <TableBody>
        {sensorData.map((data, index) => (
          <TableRow key={`dtr-${index}`}>
            <TableCell>{data.sensorId}</TableCell>
            <TableCell>{data.type}</TableCell>
            <TableCell className="text-right">{data.value}</TableCell>
            <TableCell>{data.timestamp}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  </div>;
};

export default SensorDataTable;
