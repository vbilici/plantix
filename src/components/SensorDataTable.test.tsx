
import { render, screen } from '@testing-library/react';
import SensorDataTable from "./SensorDataTable";

const mockData = [
  { sensorId: '1', type: 'temperature', value: 25, timestamp: '2023-01-01T00:00:00Z' },
];

describe("Sensor Data Table", () => {

  it('renders sensor data table correctly', async () => {
    const {container} = render(<SensorDataTable sensorData={mockData} />); 

    const sensorTypeElement = await screen.findByText('temperature');
    expect(sensorTypeElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});