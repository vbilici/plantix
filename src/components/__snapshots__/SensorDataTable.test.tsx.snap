// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sensor Data Table renders sensor data table correctly 1`] = `
<div>
  <div
    class="mx-auto"
  >
    <div
      class="tremor-Table-root overflow-auto"
    >
      <table
        class="tremor-Table-table w-full text-tremor-default text-tremor-content dark:text-dark-tremor-content"
      >
        <thead
          class="tremor-TableHead-root text-left text-tremor-content dark:text-dark-tremor-content"
        >
          <tr
            class="tremor-TableRow-row"
          >
            <th
              class="tremor-TableHeaderCell-root whitespace-nowrap text-left font-semibold top-0 px-4 py-3.5 text-tremor-content-strong dark:text-dark-tremor-content-strong"
            >
              Sensor Id
            </th>
            <th
              class="tremor-TableHeaderCell-root whitespace-nowrap text-left font-semibold top-0 px-4 py-3.5 text-tremor-content-strong dark:text-dark-tremor-content-strong"
            >
              Type
            </th>
            <th
              class="tremor-TableHeaderCell-root whitespace-nowrap font-semibold top-0 px-4 py-3.5 text-tremor-content-strong dark:text-dark-tremor-content-strong text-right"
            >
              Value
            </th>
            <th
              class="tremor-TableHeaderCell-root whitespace-nowrap text-left font-semibold top-0 px-4 py-3.5 text-tremor-content-strong dark:text-dark-tremor-content-strong"
            >
              Timestamp
            </th>
          </tr>
        </thead>
        <tbody
          class="tremor-TableBody-root align-top divide-y divide-tremor-border dark:divide-dark-tremor-border"
        >
          <tr
            class="tremor-TableRow-row"
          >
            <td
              class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4"
            >
              1
            </td>
            <td
              class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4"
            >
              temperature
            </td>
            <td
              class="tremor-TableCell-root align-middle whitespace-nowrap p-4 text-right"
            >
              25
            </td>
            <td
              class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4"
            >
              2023-01-01T00:00:00Z
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
